src/skin/colorschemeparser.cpp

```diff
@@ -11,45 +11,52 @@
 #include "skin/imgcolor.h"
 #include "skin/imginvert.h"
 #include "skin/legacyskinparser.h"
+#include "skin/skincontext.h"
 
 void ColorSchemeParser::setupLegacyColorSchemes(QDomElement docElem,
                                                 UserSettingsPointer pConfig,
-                                                QString* pStyle) {
-    QDomNode colsch = docElem.namedItem("Schemes");
+                                                QString* pStyle,
+                                                SkinContext& context) {
```

Comment:
We don't use non const references in Mixxx.

Background: 
References are pointers that allow to access the referenced value with the same syntax used for lokal variables.
Unfortunately when writing these values you don't see that the referenced value is changed elsewhere and not a local copy.
A = 1; it looks like the lokal A is changed.
If we only use const references the compiler will fail here in case of a reference.
*pA = 1; it is obvious that you access someone's else memory via a pointer.

Reply:
ok, so you're advising to use `const SkinContext& context` instead.
From what I've understood so far on that 'const' topic, we couldn't call the non-const method `SkinContext::updateVariables` anymore then?

Reply:
Yes, in general we use 
const SkinContext& context
or
SkinContext* pContext

The first one has the implicite C++ rule, that it must not become nullptr.
So you should in general avoid to make a reference from a pointer which can be a nullptr. Unfortunately this happens at the caller.

Reply:
We cannot use the const reference because we want to call `void SkinContext::updateVariables(const QDomNode& node);` which is not const.

Reply:
Yes, right. We need the pointer syntax here.

Reply:
I tried the pointer again in b5a292a but all my attempts fail for various (sometimes obvious) reasons. What's the correct syntax to call the member function `updateVariables`?